src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/ReleaseService.cs

```diff
@@ -84,14 +85,39 @@ public async Task CompletePublishing(Guid releaseVersionId, DateTime actualPubli
                 .SingleAsync(rv => rv.Id == releaseVersionId);
 
             _contentDbContext.ReleaseVersions.Update(releaseVersion);
-            releaseVersion.Published =
-                await _releaseVersionRepository.GetPublishedDate(releaseVersion.Id, actualPublishedDate);
+
+            var publishedDate = await _releaseVersionRepository.GetPublishedDate(releaseVersion.Id, actualPublishedDate);
+
+            releaseVersion.Published = publishedDate;
+
+            await UpdateReleaseFilePublishedDate(releaseVersion, publishedDate);
 
             await UpdatePublishedDataBlockVersions(releaseVersion);
 
             await _contentDbContext.SaveChangesAsync();
         }
 
+        private async Task UpdateReleaseFilePublishedDate(
+            ReleaseVersion releaseVersion,
+            DateTime publishedDate)
+        {
+            var releaseFiles = _contentDbContext.ReleaseFiles.Where(releaseFile =>
+                releaseFile.ReleaseVersionId == releaseVersion.Id &&
+                !string.IsNullOrWhiteSpace(releaseFile.Name) && // Ignore metadata files (null Name and Summary values identify them as metadata files)
+                !string.IsNullOrWhiteSpace(releaseFile.Summary));
+
+            if (releaseVersion.PreviousVersion is null)
+            {
+                await releaseFiles.ForEachAsync(releaseFile => releaseFile.Published = publishedDate);
+            }
+            else
+            {
+                await releaseFiles
+                    .Where(rf => rf.Published == null)
+                    .ForEachAsync(releaseFile => releaseFile.Published = DateTime.UtcNow);
```

Comment:
`publishedDate` is affected by whether or not users select to update the published date during sign off, so there's a chance it would be overridden - one of the requirements was to ignore this override if the files had changed and always show the current date. If there is no previous version, then the files haven't changed and we just set whatever has been determined to be the `publishedDate`.

Although new release files _should_ be null, I think it's better to ensure we're not updating any which may have been intentionally populated by some other process (whether they be in existing or future code).

Reply:
> publishedDate is affected by whether or not users select to update the published date during sign off

Ah right. So in my example `ExecuteUpdate` would have to be called with `DateTime.UtcNow`. If we called it with `publishedDate`, that could result in ReleaseFiles with null Published dates in an amendment getting the same date as the original release.

> Although new release files should be null, I think it's better to ensure we're not updating any which may have been intentionally populated by some other process (whether they be in existing or future code).

I'm inclined to think we can rely on the column being null if the published date needs updating, and don't need to be checking the previous version, and the ExecuteUpdate makes it easier to follow. But I feel this makes a big difference so feel free to leave it if you want.